### PR TITLE
Issue #3257730: Make strings translatable

### PR DESCRIPTION
**Pull request by NickDaelemans**
## Problem
A lot of strings inside the Social profile are not yet translatable.

## Solution
Wrapping all strings inside the t() function will make those strings translatable.

## Issue tracker
https://www.drupal.org/project/social/issues/3257730

## How to test
- [ ] Go to a page that has a non-translatable string (/all-groups for example, 'Join method')
- [ ] Look for that string in the interface translations
- [ ] You should find that string and be able to translate it
- [ ] See the translated string (might have to clear cache)

## Release notes
We made a bunch of strings translatable by wrapping them with the `t()` function. This means they are translatable now, but that does not mean that all strings are already translated.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by ribel** [comment]
Hi folks!
We also noticed that there are not-translatable strings, so I think it would be good to merge this one as soon as possible.

**Review by ribel — APPROVED** [review]

File: modules/custom/grequest/src/Plugin/GroupContentEnabler/GroupMembershipRequest.php [hunk 1]
```diff
@@ -76,7 +76,7 @@ protected function getGroupContentPermissions() {
 
     // Add extra permissions specific to membership group content entities.
     $permissions['request group membership'] = [
-      'title' => 'Request group membership',
+      'title' => t('Request group membership'),
```
Comment (Ressinel) [inline modules/custom/grequest/src/Plugin/GroupContentEnabler/GroupMembershipRequest.php:79]: Use `$this->t('Request group membership')`.

File: modules/social_features/social_comment_upload/src/SocialCommentUploadConfigOverride.php [hunk 2]
```diff
@@ -58,13 +58,13 @@ public function loadOverrides($names) {
             'weight' => 20,
             'format_type' => 'details',
             'format_settings' => [
-              'label' => 'Add attachment',
+              'label' => t('Add attachment'),
```
Comment (Ressinel) [inline modules/social_features/social_comment_upload/src/SocialCommentUploadConfigOverride.php:61]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_content_block/modules/social_event_content_block/src/Plugin/ContentBlock/EventContentBlock.php [hunk 3]
```diff
@@ -176,7 +176,7 @@ public function query(SelectInterface $query, array $fields) {
   public function supportedSortOptions() : array {
     return parent::supportedSortOptions() + [
       'event_date' => [
-        'label' => 'Event date',
+        'label' => t('Event date'),
```
Comment (Ressinel) [inline modules/social_features/social_content_block/modules/social_event_content_block/src/Plugin/ContentBlock/EventContentBlock.php:179]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_content_block/src/ContentBlockBase.php [hunk 4]
```diff
@@ -20,29 +20,29 @@ abstract class ContentBlockBase extends PluginBase implements ContentBlockPlugin
   public function supportedSortOptions() : array {
     return [
       'created' => [
-        'label' => 'Most recent',
+        'label' => t('Most recent'),
```
Comment (Ressinel) [inline modules/social_features/social_content_block/src/ContentBlockBase.php:23]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_group/modules/social_group_flexible_group/src/SocialGroupFlexibleGroupConfigOverride.php [hunk 5]
```diff
@@ -397,7 +397,7 @@ public function loadOverrides($names) {
       'exposed' => TRUE,
       'expose' => [
         'operator_id' => 'field_group_allowed_join_method_value_op',
-        'label' => 'Join method',
+        'label' => t('Join method'),
```
Comment (Ressinel) [inline modules/social_features/social_group/modules/social_group_flexible_group/src/SocialGroupFlexibleGroupConfigOverride.php:400]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_group/modules/social_group_request/src/SocialGroupRequestConfigOverride.php [hunk 6]
```diff
@@ -316,7 +316,7 @@ public function loadOverrides($names) {
                   'relationship' => 'none',
                   'group_type' => 'group',
                   'admin_label' => '',
-                  'label' => 'Message',
+                  'label' => t('Message'),
```
Comment (Ressinel) [inline modules/social_features/social_group/modules/social_group_request/src/SocialGroupRequestConfigOverride.php:319]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_post/src/Entity/Post.php [hunk 7]
```diff
@@ -212,15 +212,15 @@ public function getDefaultVisibilityByLabel($label, $reverse = FALSE) {
     $default_visibilities = [
       [
         'id' => 'community',
-        'label' => 'Community',
+        'label' => t('Community'),
```
Comment (Ressinel) [inline modules/social_features/social_post/src/Entity/Post.php:215]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_profile/modules/social_profile_fields/src/SocialProfileFieldsOverride.php [hunk 8]
```diff
@@ -62,7 +62,7 @@ public function loadOverrides($names) {
           ],
           'field_settings' => [
             'field_profile_nick_name' => [
-              'label' => 'Nickname',
+              'label' => t('Nickname'),
```
Comment (Ressinel) [inline modules/social_features/social_profile/modules/social_profile_fields/src/SocialProfileFieldsOverride.php:65]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

File: modules/social_features/social_tagging/src/SocialTaggingOverrides.php [hunk 9]
```diff
@@ -89,7 +89,7 @@ public function loadOverrides($names) {
           ->get('field_settings');
 
         $field_settings['social_tagging'] = [
-          'label' => 'Tags',
+          'label' => t('Tags'),
```
Comment (Ressinel) [inline modules/social_features/social_tagging/src/SocialTaggingOverrides.php:92]: Add `use StringTranslationTrait;` and use `$this->t('...')` in this file.

**Review by Ressinel — CHANGES_REQUESTED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2685-ia5b0ceyb5yxawxykteg1h02ao3fvisa.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc0e797499db652200dd7e

**Review by Ressinel — APPROVED** [review]

**Pull request merged**